File: eio2024-ev/2024-12-08-ev/kook/solution/sol_jah_ei.cpp
// vastab ainult jah/ei
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int n;
    cin >> n;

    vector<int> p(n), m(n);
    for (int i = 0; i < n; ++i) {
        cin >> p[i];
    }
    for (int i = 0; i < n; ++i) {
        cin >> m[i];
    }

    sort(p.begin(), p.end());
    sort(m.begin(), m.end());

    bool voimalik = 1;
    for (int i = 1; i < n; ++i) {
        if (m[i] * p[i-1] != m[i-1] * p[i]) {
            voimalik = 0;
            break;
        }
    }

    if (voimalik) {
        cout << "JAH\n";
    }
    else {
        cout << "EI\n";
    }

    return 0;
}

File: eio2024-ev/2024-12-08-ev/kook/solution/sol_jah_ei.py
n = int(input())
p = sorted(list(map(int, input().split())))
m = sorted(list(map(int, input().split())))

voimalik = True
for i in range(1,n):
    if (m[i]*p[i-1] != m[i-1]*p[i]):
        voimalik = False
    
if (voimalik):
    print('JAH')
else:
    print('EI')
